app/src/main/java/com/github/se/icebreakrr/ui/sections/Filter.kt

```diff
@@ -266,8 +286,34 @@ fun FilterScreen(
           Box(modifier = Modifier.fillMaxWidth().padding(16.dp)) {
             Button(
                 onClick = {
+                  filterViewModel.setFilteredTags(tagsViewModel.filteringTags.value)
                   tagsViewModel.applyFilters()
                   tagsViewModel.leaveUI()
+
+                  val selectedGenders = mutableListOf<Gender>()
+                  if (manSelected) selectedGenders.add(Gender.MEN)
+                  if (womanSelected) selectedGenders.add(Gender.WOMEN)
+                  if (otherSelected) selectedGenders.add(Gender.OTHER)
+                  filterViewModel.setGenders(selectedGenders)
+
+                  when {
+                    ageFrom != null && ageTo != null && !ageRangeError -> {
+                      filterViewModel.setAgeRange(ageFrom!!..ageTo!!)
+                    }
+                    ageFrom == null && ageTo == null -> {
+                      filterViewModel.setAgeRange(null)
+                    }
+                    ageFrom != null && ageTo == null -> {
+                      filterViewModel.setAgeRange(ageFrom!!..Int.MAX_VALUE)
+                    }
+                    ageFrom == null && ageTo != null -> {
+                      filterViewModel.setAgeRange(0..ageTo!!)
+                    }
+                    else -> {
+                      filterViewModel.setAgeRange(null)
+                    }
+                  }
+
                   navigationActions.goBack()
```

Comment:
You need to call getFilteredProfilesInRadius() with the updated filters to refresh the filteredProfiles list and recompose the UI. For now, only manual updates through the pull-to-refresh process will refresh the list, which is not the expected behavior.

Reply:
You're right, I'll fix it.